packages/application-generic/src/usecases/conditions-filter/conditions-filter.usecase.ts

```diff
@@ -1,27 +1,77 @@
 import { Injectable } from '@nestjs/common';
-import { StepFilter } from '@novu/dal';
-import { FilterParts, FilterPartTypeEnum, ICondition } from '@novu/shared';
+import axios from 'axios';
+import {
+  EnvironmentRepository,
+  ExecutionDetailsRepository,
+  MessageRepository,
+  StepFilter,
+  SubscriberEntity,
+  SubscriberRepository,
+  JobRepository,
+  TenantEntity,
+  TenantRepository,
+} from '@novu/dal';
+import {
+  ChannelTypeEnum,
+  FILTER_TO_LABEL,
+  FilterParts,
+  FilterPartTypeEnum,
+  ICondition,
+  IOnlineInLastFilterPart,
+  IPreviousStepFilterPart,
+  IRealtimeOnlineFilterPart,
+  IWebhookFilterPart,
+  PreviousStepTypeEnum,
+  TimeOperatorEnum,
+  IMessageFilter,
+  FieldOperatorEnum,
+  FieldLogicalOperatorEnum,
+} from '@novu/shared';
 import { Filter } from '../../utils/filter';
 import {
   FilterProcessingDetails,
   IFilterVariables,
 } from '../../utils/filter-processing-details';
 import { ConditionsFilterCommand } from './conditions-filter.command';
+import {
+  differenceInDays,
+  differenceInHours,
+  differenceInMinutes,
+  parseISO,
+} from 'date-fns';
+import { EmailEventStatusEnum } from '@novu/stateless';
+import { PlatformException } from '../../utils/exceptions';
+import { createHash } from '../../utils/hmac';
+import { Instrument } from '../../instrumentation';
+import { CachedEntity } from '../../services/cache/interceptors/cached-entity.interceptor';
+import { buildSubscriberKey } from '../../services/cache/key-builders/entities';
 
 @Injectable()
 export class ConditionsFilter extends Filter {
-  public async filter(
-    command: ConditionsFilterCommand,
-    variables: IFilterVariables
-  ): Promise<{
+  constructor(
```

Comment:
This is one of my main concerns and what I tried to tackle in my PR (https://github.com/novuhq/novu/pull/4702).
Most of the logic here inside should be a service and try to make it reusable. This neither follows the Execution pattern we have implemented in other use cases.